app/models/account_balance.rb

```diff
@@ -1,6 +1,8 @@
 class AccountBalance < ApplicationRecord
   belongs_to :account
 
+  validates :account_id, :date, :balance, presence: true
```

Comment:
In this case, I think we should remove the `account_id` validation.  We'll often be doing bulk operations on this table and validating the association (`:account`) adds a DB query that will already be enforced at the DB level but will make these bulk operations less performant

Reply:
In the case of bulk operations, I would recommend passing `validate: false` to the mutative db methods during save or update, which skips validations. Alternatively, and somewhat more dangerously you could use `upsert` or other direct `INSERT` methods which avoid validations and model instantiation entirely. 

But I'm happy to make this change to help make this mergable.

Reply:
Yep, those are possible workarounds.  Up to you—if keeping it, just make sure to update to `:account` to get the full validation check.

Reply:
Thanks Zach! If you are happy either way on validating the `:account` for now, then I think all comments are addressed.